### DISCUSSION TITLE
Does Swell support videos for Product(s)?

### DISCUSSION THREAD

**Discussion by waldothedeveloper** [swellstores/community#534] [Help] [answered]
I've been trying to find anything related to 'upload' videos in the Swell documentation, but I don't find anything at all. I guess is not currently supported?

Going with this assumption, what is the alternative to allow having videos "stored" for Product(s) just like images? 

For example, Etsy allows vendors to store up to 10 images and 1 video. I'm building the storefront with NextJS (app router), so I would like to have a similar idea to Etsy. Allow my vendors to uplaod up to 10 images and 1 video per Product (or Service). 

What's the best approach for this? 

What I'm thinking to solve this is creating a custom property in the Product Model to store the URL(s) of the uploaded videos that can be stored in something like Amazon S3, Firebase storage, Mux or similar. 

I would like to get some technical opinions about this approach from the community and/or the Swell developers team. 

Appreciate it

**Comment by logeshswell** [comment]
You can create a new custom field of the type "Video" as follows:
<img width="695" alt="Screenshot 2024-05-15 at 6 46 18 PM" src="https://github.com/swellstores/community/assets/100766131/92a37423-e1a9-4fe5-ac12-538ca772d36b">

**Reply by waldothedeveloper** [reply]
Hey @logeshswell, thanks for the quick reply. I did what you recommended me. But there seems to be a hard limit on the size of videos that can be uploaded. I would like to have more flexibility on this though. 

![image](https://github.com/swellstores/community/assets/16472274/9b366c6e-8fd1-4f86-8284-005c79a34fb7)

Could you please tell me if it's possible to create a custom field of type Array where I can store URLs of videos that I can upload to S3, Mux, or Firebase storage?

**Reply by logeshswell** [reply, marked as answer]
Yes, you can create a custom field of type **Tags** to store an array of URLs.
